packages/shared/src/components/cards/ReferralWidget.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { ReactElement } from 'react';
+import { Button, ButtonSize } from '../buttons/Button';
+import {
+  getFacebookShareLink,
+  getTwitterShareLink,
+  getWhatsappShareLink,
+} from '../../lib/share';
+import WhatsappIcon from '../icons/Whatsapp';
+import FacebookIcon from '../icons/Facebook';
+import TwitterIcon from '../icons/Twitter';
+import { TextField } from '../fields/TextField';
+import { ReferralCampaignKey, useReferralCampaign } from '../../hooks';
+import { link } from '../../lib/links';
+import { useShareOrCopyLink } from '../../hooks/useShareOrCopyLink';
+import { labels } from '../../lib';
+import { AnalyticsEvent, TargetId } from '../../lib/analytics';
+import { SimpleTooltip } from '../tooltips';
+
+const ReferralWidget = (): ReactElement => {
+  const { isReady, url } = useReferralCampaign({
+    campaignKey: ReferralCampaignKey.Generic,
+  });
+  const inviteLink = url || link.referral.defaultUrl;
+  const [, onShareOrCopyLink] = useShareOrCopyLink({
+    text: labels.referral.generic.inviteText,
+    link: inviteLink,
+    trackObject: () => ({
+      event_name: AnalyticsEvent.CopyReferralLink,
+      target_id: TargetId.ProfilePage,
+    }),
+  });
+  const onShareClick = () => {
+    onShareOrCopyLink();
+  };
+
+  return (
+    <div className="flex laptopL:absolute flex-col tablet:flex-row laptopL:flex-col flex-wrap justify-between p-4 laptopL:m-6 mt-6 laptopL:mt-0 mb-4 laptopL:max-w-widget laptopL:h-auto rounded-2xl border border-theme-divider-tertiary laptopL:left-[40rem] max-w-fit bg-theme-bg-primary">
+      <h3 className="mb-2 font-bold typo-title3">Invite friends</h3>
+      <p className="contents text-salt-50 typo-callout">
+        Tell your dev friends how easy is it to learn, collaborate, and grow
+        together
+      </p>
+      <div className="flex flex-col my-5 w-auto tablet:w-70">
```

Comment:
It looks more like this is always rendered 100% of the parent no?

Reply:
Hmm, no not always, image below shows where it's not 100% of parent in tablet + the width for laptop and above of the whole widget is defined by the width of this element, that w-70. Unless u meant something else... 🤔 
<img width="595" alt="Screenshot 2023-11-13 at 7 10 10 AM" src="https://github.com/dailydotdev/apps/assets/36197304/a0def776-85f1-4fc7-9005-6acccab91ed5">

Reply:
I think it's more of a design flaw to be honest.
We should just have the minimal space requirement between the elements the rest should hit automatically I think.

Reply:
Hm, the link and invite are not within a separate tag with the other header and description, which changes a bit on how to arrange the elements, if they were in a separate tag then yeah what you are saying would be doable. I just didn't want to introduce another tag when we could do with what we have. Or maybe I'm not envisioning properly how to do this 🤔 
What would be wrong with what we have already?

Reply:
Agreed to keep this as is